public/js/services/cart-service.js

```diff
@@ -8,6 +8,35 @@ class CartService {
       ? JSON.parse(localStorage.getItem('cart')).length
       : 0
   }
+
+  isCartEmpty() {
+    return !localStorage.getItem('cart') ? true : false
+  }
+
+  addToCart(id, quantity) {
+    if (this.isCartEmpty()) {
+      localStorage.setItem('cart', JSON.stringify([{ id, quantity }]))
+    } else {
+      var cart = this.getCartProducts()
+      var index = cart.findIndex((x) => x.id == id)
+      if (index !== -1) {
+        cart[index].quantity = cart[index].quantity + quantity
+      } else {
+        cart.push({ id, quantity })
+      }
+      localStorage.setItem('cart', JSON.stringify(cart))
+    }
+    this.updateCart()
+  }
+
+  updateCart() {
+    var cartCount = this.getCartItemsCount()
+    if (cartCount) {
+      $('#cart-count').show().text(cartCount)
+    } else {
+      $('#cartCount').hide()
+    }
+  }
```

Comment:
It'd better to keep all UI logic out of the service files. Service files only deal with APIs and data. Other JS files call on services and do the UI stuff themselves.

But you probably moved the code here so that we can update the cart count dynamically (on add-to-cart). In that case, this looks like a good place to introduce Event handling so that we can decouple things safely.

```suggestion
  updateCart() {
    document.dispatch(new Event("cartUpdate"));
  }
  
  onCartUpdate(fn) {
    document.addEventListener("cartUpdate", fn)
  }
```

and `common.js` do something like

```js
function updateCartCounter() {
  var cartCount = cartService.getCartItemsCount()
  if (cartCount) {
    $('#cart-count').show().text(cartCount)
  } else {
    $('#cartCount').hide()
  }
}

cartService.onCartUpdate(updateCartCount)
updateCartCount()
```

Let me know if there's something i've overlooked though